test/shape.cpp

```diff
@@ -526,6 +526,12 @@ TEST(shape_make_compact) {
   assert_shapes_eq(make_compact(s6), s6_compact);
 }
 
+TEST(shape_fixed_dense) { 
+  shape<dim<0, 2, 1>, dim<0, 3, 2>, dim<0, 4, 6>> s1;
+  fixed_dense_shape<2, 3, 4> s1_fixed;
```

Comment:
Maybe make this 4-dimensional to reduce the chance that the reader gets confused - I initially thought these might be (min, extent, stride) instead of all extents.